validate.go

```diff
@@ -24,6 +24,60 @@ func adjustResourceRequest(container *corev1.Container, resourceName string, res
 	return false
 }
 
+func validateContainerResourceLimits(container *corev1.Container, settings *Settings) error {
+	if container.Resources.Limits == nil && settings.Cpu.IgnoreValues && settings.Memory.IgnoreValues {
+		return fmt.Errorf("container does not have any resource limits")
+	}
+
+	cpuStr, found := container.Resources.Limits["cpu"]
+	if settings.Cpu.IgnoreValues && (!found || cpuStr == nil || len(strings.TrimSpace(string(*cpuStr))) == 0) {
+		return fmt.Errorf("container does not have a cpu limit")
+	}
+
+	memoryStr, found := container.Resources.Limits["memory"]
+	if settings.Memory.IgnoreValues && (!found || memoryStr == nil || len(strings.TrimSpace(string(*memoryStr))) == 0) {
```

Comment:
We use the same piece of code to check if there is something in the resource at 3 different places. Let's extract that into a function. Like this:

```suggestion
	if settings.Memory.IgnoreValues && missingResourceQuantity(container.Resource.Limits, "memory"){
```
The function code could be something like:

```go
func missingResourceQuantity(resources map[string]*apimachinery_pkg_api_resource.Quantity, resourceName string) bool{

resourceStr, found := resources[resourceName]
return !found || resourceStr == nil || len(strings.TrimSpace(string(*resourceStr))) == 0

}
```

Call the suggested function in the `adjustResourceRequest` function as well.

Reply:
This is taken care of here. https://github.com/kubewarden/container-resources-policy/pull/18/commits/f68a538ec29feb1afe0ffdd995e96f67f4e2f4f2